runtime/src/commonMain/kotlin/app/cash/sqldelight/db/QueryResult.kt

```diff
@@ -17,6 +17,8 @@ sealed interface QueryResult<T> {
   suspend fun await(): T
 
   data class Value<T>(override val value: T) : QueryResult<T> {
+    constructor(block: () -> T) : this(block())
```

Comment:
i think I understand why we added this, but could it be avoided if the transaction APIs were `QueryResult<() -> Unit>` and the result was also invoked when awaited/accessed?

Reply:
makes sense, in general my default is always for less stuff in the public API